libs/go/libp2p_node/dht/dhtclient/dhtclient.go

```diff
@@ -66,6 +66,7 @@ const (
 	bootstrapTimeout          = 1 * 60 * time.Second // doesn't include peer restart
 	sleepTimeDefaultDuration  = 100 * time.Millisecond
 	sleepTimeIncreaseMFactor  = 2 // multiplicative increase
+	reconnectTimeout          = 5 * time.Second
```

Comment:
This is new; can you elaborate?

Reply:
in the reconnect code there is no timeout, so it can stuck, what actually happened. after timeout context added it start work as expected.